lib/diagnostic.rb

```diff
@@ -7,36 +7,38 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.select { |num| }
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce do |accumulator, element|
+      accumulator + element
+    end
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any? { |n| n.odd == true }
   end
```

Comment:
`.odd` is not a method that exists in Ruby. I think what you intended was this: https://ruby-doc.org/core-2.2.2/Integer.html#method-i-odd-3F

Notice that this method immediately returns a `true` or `false`. While `true == true` does evaluate to `true`, it's a bit redundant. Consider this possible solution:
```rb
  def some_odd(numbers)
    numbers.any?(&:odd?)
  end
```